cmake/LibraryConfigurations.cmake

```diff
@@ -0,0 +1,109 @@
+
+### Config options ###
+# Translate the config options to what raylib wants
+configure_file(config.h.in ${CMAKE_BINARY_DIR}/cmake/config.h)
+
+if(${PLATFORM} MATCHES "Desktop")
+    set(PLATFORM_CPP "PLATFORM_DESKTOP")
+    
+    if(APPLE)
+        # Need to force OpenGL 3.3 on OS X
+        # See: https://github.com/raysan5/raylib/issues/341
+        set(GRAPHICS "GRAPHICS_API_OPENGL_33")
+        find_library(OPENGL_LIBRARY OpenGL)
+        set(LIBS_PRIVATE ${OPENGL_LIBRARY})
+        link_libraries("${LIBS_PRIVATE}")
+        if (NOT CMAKE_SYSTEM STRLESS "Darwin-18.0.0")
+            add_definitions(-DGL_SILENCE_DEPRECATION)
+            MESSAGE(AUTHOR_WARNING "OpenGL is deprecated starting with macOS 10.14 (Mojave)!")
+        endif()
+    elseif(WIN32)
+        add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+        set(LIBS_PRIVATE ${LIBS_PRIVATE} winmm)
+    else()
+        find_library(pthread NAMES pthread)
+        find_package(OpenGL QUIET)
+        if ("${OPENGL_LIBRARIES}" STREQUAL "")
+            set(OPENGL_LIBRARIES "GL")
+        endif()
+        
+        if ("${CMAKE_SYSTEM_NAME}" MATCHES "(Net|Open)BSD")
+            find_library(OSS_LIBRARY ossaudio)
+        endif()
+        
+        set(LIBS_PRIVATE m pthread ${OPENGL_LIBRARIES} ${OSS_LIBRARY})
+    endif()
+
+elseif(${PLATFORM} MATCHES "Web")
+    set(PLATFORM_CPP "PLATFORM_WEB")
+    set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
+    set(CMAKE_C_FLAGS "-s USE_GLFW=3 -s ASSERTIONS=1 --profiling")
+    set(CMAKE_STATIC_LIBRARY_SUFFIX ".a")
+
+elseif(${PLATFORM} MATCHES "Android")
+    set(PLATFORM_CPP "PLATFORM_ANDROID")
+    set(GRAPHICS "GRAPHICS_API_OPENGL_ES2")
+    include(AddIfFlagCompiles)
+    add_if_flag_compiles(-ffunction-sections CMAKE_C_FLAGS)
+    add_if_flag_compiles(-funwind-tables CMAKE_C_FLAGS)
```

Comment:
Messing around with compiler flags belong into the `CMAKE_TOOLCHAIN_FILE`.

Reply:
Out of the scope of this pull request @Neumann-A there will be more refactoring in the future. There is a lot of messing with compiler flags and not only in this file but the current aim is to size down the cmake system into a more consumable chunks.